domain/model/state/state.go

```diff
@@ -157,6 +157,109 @@ func (s *State) GetModel(ctx context.Context, uuid coremodel.UUID) (coremodel.Mo
 	})
 }
 
+// GetModelByName returns the model found for the given username and model name
+// for which there can only be one. Should no model be found for the provided
+// search criteria an error satisfying [modelerrors.NotFound] will be returned.
+func (s *State) GetModelByName(
+	ctx context.Context,
+	username string,
+	modelName string,
+) (coremodel.Model, error) {
+	db, err := s.DB()
+	if err != nil {
+		return coremodel.Model{}, errors.Trace(err)
+	}
+
+	q := `
+SELECT uuid,
+       name,
+       ma.target_version           AS agent_version,
+       cloud_name,
+       cloud_region_name,
+       model_type,
+       owner_uuid,
+	   owner_name,
+       cloud_credential_cloud_name,
+       cloud_credential_owner_name,
+       cloud_credential_name,
+       life
+FROM v_model
+INNER JOIN model_agent ma ON v_model.uuid = ma.model_uuid
+WHERE name = ?
+AND owner_name = ?
+`
+
+	var (
+		agentVersion string
+		cloudRegion  sql.NullString
+		modelType    string
+		userUUID     string
+		credName     sql.NullString
+		credOwner    sql.NullString
+		credCloud    sql.NullString
+		model        coremodel.Model
+		modelIdStr   string
+	)
+
+	err = db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
```

Comment:
Use sqlair...

Reply:
Cards are made. Will tackle in the model epic. Haven't forgotten about them.